src/main/java/com/algorand/algosdk/crypto/LogicsigSignature.java

```diff
@@ -124,11 +120,9 @@ public boolean verify(Address singleSigner) throws NoSuchAlgorithmException {
             return false;
         }
 
-        try {
-            Logic.checkProgram(this.logic, this.args);
-        } catch (Exception ex) {
-            return false;
-        }
+        // NOTE: Logic.checkProgram has been deprecated, use Logic.sanityCheckProgram instead
+        //       we no longer read from langspec.json, we do not need to catch IOException error
+        Logic.sanityCheckProgram(this.logic);
```

Comment:
@ahangsu Minor and optional:  For replacements _not_ exposed publicly, I don't think an inline citation is required.  If you prefer it, feel welcomed to keep as is.

```suggestion
        Logic.sanityCheckProgram(this.logic);
```